### ISSUE TITLE
Make Part 1 Core Mandatory

### ISSUE THREAD

**Issue by cmheazel** [opengeospatial/ogcapi-common#283]
The requirements in Part 1 Core address the following:

 1 -Use HTTP or HTTPS
 2 - How to handle unknown query parameter names
 3 - How to handle invalid query parameter values
 4 - Formatting rules governing query parameter names and values.

Number 1 should apply to any Web API.
Numbers 2, 3, and 4 should apply to any OGC Web API standard

Yet we have decided that Part 1 Core is not mandatory.  So how do we flow these requirements down to where they are needed?

**Comment by cportele** [comment]
I think this goes back to the discussion during the June Open OAB meeting.

What is wrong with Web APIs that conform to specific OGC API conformance classes without being bound by certain OGC API conventions like items 2 to 4?

A Web API that supports them can declare this fact to clients in the Conformance Declaration, if the API is more OGC-centric and supports the Landing Page conformance class, too.

A key goal of the OGC API standards is or at least was to provide guidance how to spatially enable Web APIs, not try to turn all Web APIs that use spatial information into Web APIs that are centred around OGC concepts. 

I think we have to realize that we haven't been doing a great job towards this goal and we have to decide, if we want to improve how we position the OGC API building blocks and the OGC API standards. We really should make it simpler for others to find and implement the key geospatial OGC API building blocks (like the bbox query parameter) without fully subscribing to all the other OGC API concepts.

I don't think the current Common does this. It is largely focused on an OGC-centric as well as a document-centric view and mainly provides guidance to OGC API standards editors. That can be valuable, too, but it is a narrower focus.

My conclusion from the discussions over the last year or so is that we really should or have to position/market/publish the key geospatial OGC API building blocks separately from the standards / conformance classes in which they are specified. Common could have played that role, but it doesn't seem like this is the current direction.

**Comment by cmheazel** [comment]
@cportele At what point does a module become too small to be useful?
Let's talk about the [Bounding Box module](https://github.com/opengeospatial/ogcapi-common/tree/master/api_modules/bbox). This is a reusable Requirements Module which defines the `bbox` parameter and provides general rules on how it should be processed. This is useless as a software module because it doesn't specify the operation (HTTP verb) or target resource (Path). It is useful for a standards developer because it allows them to re-use the concept of a `bbox` parameter for their context (operation and resource). Now, if I were to create a `Simple Query` Requirements Class which marries the `bbox` Requirements module with the HTTP GET verb and an `extent` resource, then I have a reusable software module. 

I don't think we can slice it much finer than that.

**Comment by cportele** [comment]
@cmheazel - The `bbox` building block can be useful for any API designer like it is for an OGC API standards developer. Just like we bind it to GET requests on the Features resource in OGC API Features, anyone can bind it to their API resources / operations even if they do not use anything else from the OGC API building blocks in their API.

PS: "module" is an alias for "[API] building block"?

**Comment by cmheazel** [comment]
@cportele I can't speak for any designer since our scope is OGC.  But if they find it useful why not?
The Requirements Module concept is taken from the mod spec. Trying to stay orthodox here.
I would divide "API Building Block" into two categories:
 - API Standard Building Block = Requirements Module
 - API Software Building Block = Requirements Class
The difference is that a Requirements Module does not completely define an implementable capability. It provides a component without context, where context = parameter, resource, and operation.

**Comment by cmheazel** [comment]
@cportele I agree in principle. But we don't yet know what that new "format" will look like.  There is a thread in TB-17 looking into that question. I've been re-structuring the GitHub in anticipation of the results. 

The two categories of building blocks are derived from my attempts to implement building blocks in Common Part 2. When looking at a module I ask two question:
1) Is it useful?
2) Can I implement it in a Java class?
It turns out that a lot of useful concepts are not defined well enough to implement as a software module. But they are still useful for building standards (and enforcing a common look and feel). Thus the two categories.

**Comment by cmheazel** [comment]
The current version of API-Common Part 2 - `/Collections` requirements class does not require anything from Part 1, but it does require HTTP (citing RFC-7231).  Implementation of Part 1 Core and Landing Page is recommended but not required.

**Comment by jerstlouis** [comment]
Common - Part 2: Geospatial Data "Collections" requirement class now depends on Comon - Part 1 "Core" requirement class.

This is only the general principles, and does not include the Landing Page, Conformance and API definition requirements, which are specified in the "Landing Page" requirements class.

If there is anything too restrictive in the Core requirements of Part 1: Core, then that probably needs to be fixed in a revision.
